packages/flutter/lib/src/semantics/semantics.dart

```diff
@@ -1318,15 +1318,21 @@ class SemanticsNode extends AbstractNode with DiagnosticableTreeMixin {
   void _addToUpdate(ui.SemanticsUpdateBuilder builder) {
     assert(_dirty);
     final SemanticsData data = getSemanticsData();
-    Int32List children;
+    Int32List childrenInTraversalOrder;
+    Int32List childrenInHitTestOrder;
     if (!hasChildren || mergeAllDescendantsIntoThisNode) {
-      children = _kEmptyChildList;
+      childrenInTraversalOrder = _kEmptyChildList;
+      childrenInHitTestOrder = _kEmptyChildList;
     } else {
+      final int childCount = _children.length;
       final List<SemanticsNode> sortedChildren = _childrenInTraversalOrder();
-      final int childCount = sortedChildren.length;
-      children = new Int32List(childCount);
-      for (int i = 0; i < childCount; ++i) {
-        children[i] = sortedChildren[i].id;
+      childrenInTraversalOrder = new Int32List(childCount);
+      for (int i = 0; i < childCount; i += 1) {
+        childrenInTraversalOrder[i] = sortedChildren[i].id;
+      }
+      childrenInHitTestOrder = new Int32List(childCount);
```

Comment:
Maybe document here for future code readers that `_children` is in paint order and inverting it gives us hit test order?